localstack/services/apigateway/models.py

```diff
@@ -34,6 +34,8 @@ class RestApiContainer:
     models: Dict[str, Model]
     # maps ResourceId of a Resource to its children ResourceIds
     resource_children: Dict[str, List[str]]
+    # maps ModelId to Resource methods using it
+    models_in_use: Dict[str, List[str]]
```

Comment:
True, I had thought about it too. We would need to iterate over every `Resource` which then holds several `Method` and check each for its `RequestModels`. It's doable I suppose, and the performance cost should not be too high. It would make things much simpler than keeping mappings, that's true. I'll give it a try! 👍

Reply:
We will fail parity on the last Method to use the Model, but I guess that's fine, I'll add a `skip_snapshot_verify` path.